src/Components/WebAssembly/WebAssembly.Authentication/src/Services/AuthorizationMessageHandler.cs

```diff
@@ -0,0 +1,123 @@
+        /// <param name="navigation">The <see cref="NavigationManager"/> to use for performing redirections.</param>
+        public AuthorizationMessageHandler(
+            IAccessTokenProvider provider,
+            NavigationManager navigation)
+        {
+            _provider = provider;
+            _navigation = navigation;
+        }
+
+        /// <inheritdoc />
+        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            var now = DateTimeOffset.Now;
+            if (_allowedUris == null)
+            {
+                throw new InvalidOperationException($"The '{nameof(AuthorizationMessageHandler)}' is not configured. " +
+                    $"Call '{nameof(AuthorizationMessageHandler.ConfigureHandler)}' and provide a list of endpoint urls to attach the token to.");
+            }
+            if (_allowedUris.Any(uri => uri.IsBaseOf(request.RequestUri)))
+            {
+                if (_lastToken == null || now >= _lastToken.Expires.AddMinutes(-5))
+                {
+                    var tokenResult = _tokenOptions != null ?
+                        await _provider.RequestAccessToken(_tokenOptions) :
+                        await _provider.RequestAccessToken();
+
+                    if (tokenResult.TryGetToken(out var token))
+                    {
+                        _lastToken = token;
+                        _cachedHeader = new AuthenticationHeaderValue("Bearer", _lastToken.Value);
+                    }
+                    else
+                    {
+                        throw new AccessTokenNotAvailableException(tokenResult, _navigation);
+                    }
+
+                    // We don't try to handle 401s and retry the request with a new token automatically since that would mean we need to copy the request
+                    // headers and buffer the body and we expect that the user instead handles the 401s. (Also, we can't really handle all 401s as we might
+                    // not be able to provision a token without user interaction).
+                    request.Headers.Authorization = _cachedHeader;
+                }
+            }
+
+            return await base.SendAsync(request, cancellationToken);
+        }
+
+        /// <summary>
+        /// Configures this handler to attach the token to the given list of urls and optionally request the given scopes and use the given return url
+        /// instead of the defaults provided by the authentication implementation.
+        /// </summary>
+        /// <param name="endpointUrls">The list of base addresses of endpoint urls to which the token will be attached to.</param>
+        /// <param name="scopes">The list of scopes to use when requesting a token to attach to outgoing requests.</param>
+        /// <param name="returnUrl">The return url to use in case there is an issue provisioning the token and a redirection to the
```

Comment:
```suggestion
        /// <param name="returnUrl">The return URL to use in case there is an issue provisioning the token and a redirection to the
```

Reply:
done